src/tools/tidy/src/ui_tests.rs

```diff
@@ -1,9 +1,42 @@
-//! Tidy check to ensure that there are no stray `.stderr` files in UI test directories.
+//! Tidy check to ensure:
+//! - the number of entries in each directory must be less than 1000
+//! - there are no stray `.stderr` files
+//! in UI test directories.
 
 use std::fs;
 use std::path::Path;
 
+const ENTRY_LIMIT: usize = 1000;
+
+fn check_entries(path: &Path, bad: &mut bool) {
+    let dirs = walkdir::WalkDir::new(&path.join("test/ui"))
+        .into_iter()
+        .filter_entry(|e| e.file_type().is_dir());
+    // FIXME: We should not allow them ideally.
+    let allowed_paths = ["test/ui", "test/ui/issues"];
+    for dir in dirs {
+        if let Ok(dir) = dir {
+            let dir_path = dir.path();
+            let is_allowed = allowed_paths.iter().any(|p| path.join(p) == dir_path);
+            if is_allowed {
+                continue;
+            }
+            let count = std::fs::read_dir(dir_path).unwrap().count();
+            if count >= ENTRY_LIMIT {
+                tidy_error!(
+                    bad,
+                    "following path contains more than {} entries (current: {}): {}",
```

Comment:
If this error is reported for `test/ui` or `test/ui/issues` there should also a suggestion to move the test to some relevant subdirectory.